packages/browser-wallet/src/popup/pages/Web3ProofRequest/utils.ts

```diff
@@ -0,0 +1,148 @@
+
+export type SecretStatementV2 = Exclude<AtomicStatementV2, RevealStatementV2>;
+
+export interface DisplayCredentialStatementProps<Statement> extends ClassName {
+    credentialStatement: Statement;
+    dappName: string;
+    setChosenId: (id: string) => void;
+    net: Network;
+}
+
+/**
+ * Build the commitmentInputs required to create a presentation for the given statement.
+ */
+export function getCommitmentInput(
+    statement: RequestStatement,
+    wallet: ConcordiumHdWallet,
+    identities: ConfirmedIdentity[],
+    credentials: WalletCredential[],
+    verifiableCredentials: VerifiableCredential[]
+): CommitmentInput {
+    if (statement.type) {
+        const cred = verifiableCredentials?.find((c) => c.id === statement.id);
+
+        if (!cred) {
+            throw new Error('IdQualifier not fulfilled');
+        }
+
+        return createWeb3CommitmentInputWithHdWallet(
+            wallet,
+            getContractAddressFromIssuerDID(cred.issuer),
+            cred.index,
+            cred.credentialSubject,
+            cred.randomness,
+            cred.signature
+        );
+    }
+    const credId = getCredentialIdFromSubjectDID(statement.id);
+    const credential = credentials.find((cred) => cred.credId === credId);
+
+    if (!credential) {
+        throw new Error('IdQualifier not fulfilled');
+    }
+
+    const identity = (identities || []).find(isIdentityOfCredential);
+
+    if (!identity || identity.status !== CreationStatus.Confirmed) {
+        throw new Error('No identity found for credential');
+    }
+
+    return createAccountCommitmentInputWithHdWallet(
+        statement.statement,
+        identity.providerIndex,
+        identity.idObject.value.attributeList,
+        wallet,
+        identity.providerIndex,
+        credential.credNumber
+    );
+}
+
+/**
+ * Given a credential statement for an account credential, and a list of account credentials, return the filtered list of credentials that satisfy the statement.
+ * Note this also requires the identities for the account credentials as an additional argument, to actually check the attributes of the credential.
+ */
+export function getViableAccountCredentialsForStatement(
+    credentialStatement: AccountCredentialStatement,
+    identities: ConfirmedIdentity[],
+    credentials: WalletCredential[]
+): WalletCredential[] {
+    const allowedIssuers = credentialStatement.idQualifier.issuers;
+    return credentials?.filter((c) => {
+        if (allowedIssuers.includes(c.providerIndex)) {
+            const identity = (identities || []).find((id) => isIdentityOfCredential(id)(c));
+            if (identity && identity.status === CreationStatus.Confirmed) {
+                return canProveCredentialStatement(credentialStatement, identity.idObject.value.attributeList);
+            }
+        }
+        return false;
+    });
+}
+
+function doesCredentialSatisfyStatement(statement: AtomicStatementV2, cred: VerifiableCredential): boolean {
```

Comment:
Basically the same, the `canProveCredentialStatement'`s arguments are just awkward. 
I'll fix it in the SDK so it can be reused + added a TODO here to replace the function.